saleor/order/migrations/tasks/saleor3_13.py

```diff
@@ -46,3 +51,63 @@ def order_events_rename_transaction_capture_events_task():
     if ids:
         update_type_to_transaction_charge_requested(qs)
         order_events_rename_transaction_capture_events_task.delay()
+
+
+@app.task
+def update_orders_charge_statuses_task():
+    orders = Order.objects.filter(
+        Exists(OrderGrantedRefund.objects.filter(order_id=OuterRef("pk")))
+    )
+    orders = orders.annotate(
+        refunded_amount=Coalesce(
+            Sum("granted_refunds__amount_value"),
+            0,
+            output_field=DecimalField(),
+        ),
+        current_total_gross=F("total_gross_amount") - F("refunded_amount"),
+    )
+    trigger_task = False
+
+    overcharged_orders = orders.filter(
+        total_charged_amount__gt=F("current_total_gross")
+    ).exclude(charge_status=OrderChargeStatus.OVERCHARGED)[:BATCH_SIZE]
```

Comment:
I wonder if hitting database every time for each case is correct approach here. `orders` query above has some cost already. If we just load `orders[:BATCH_SIZE]` into memory and iterate through it in python to deduce witch `order_id`'s should use to update it may be safer if I'm not wrong. WDYT?